addons/mail/static/src/core/common/thread.js

```diff
@@ -310,10 +319,18 @@ export class Thread extends Component {
                 !this.env.messageHighlight?.highlightedMessageId &&
                 thread.scrollTop !== undefined
             ) {
-                const value =
-                    thread.scrollTop === "bottom"
-                        ? ref.el.scrollHeight - ref.el.clientHeight
-                        : thread.scrollTop;
+                let value;
+                if (this.props.order === "asc") {
+                    value =
+                        thread.scrollTop === "bottom"
+                            ? ref.el.scrollHeight - ref.el.clientHeight
+                            : thread.scrollTop;
+                } else {
+                    value =
+                        thread.scrollTop === "bottom"
+                            ? 0
+                            : ref.el.scrollHeight - thread.scrollTop - ref.el.clientHeight;
+                }
```

Comment:
I feel that putting the `thread.scrollTop` condition at the top level ease the understanding of this flow. Also it is very strange to store "bottom" as the `scrollTop` value but to set `scrollTop` to `0` in case the order is "desc". Its quite the opposite^^ Couldn't we save the proper value instead and keep the restore code intact?
```suggestion
let value;
if (thread.scrollTop === "bottom") {
    value = this.props.order === "asc" ? ref.el.scrollHeight - ref.el.clientHeight : 0;
} else {
    value = this.props.order === "asc" ? thread.scrollTop : ref.el.scrollHeight - thread.scrollTop - ref.el.clientHeight;
}
```